client/my-sites/site-settings/disconnect-site/index.jsx

```diff
@@ -9,49 +9,40 @@ import { localize } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import DisconnectSurvey from './disconnect-survey';
-import DocumentHead from 'components/data/document-head';
-import FormattedHeader from 'components/formatted-header';
-import Main from 'components/main';
-import NavigationLink from 'components/wizard/navigation-link';
+import { DownFlow } from './down-flow';
 import redirectNonJetpack from 'my-sites/site-settings/redirect-non-jetpack';
-import Troubleshoot from './troubleshoot';
-import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { SurveyFlow } from './survey-flow';
+import { getSelectedSite } from 'state/ui/selectors';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-const DisconnectSite = ( { reason, siteSlug, translate } ) => {
-	const confirmHref = '/settings/disconnect-site/confirm/' + siteSlug;
+const DisconnectSite = ( { reason, type, site, translate } ) => {
+	const confirmHref = `/settings/disconnect-site/confirm/${ site.slug }`;
 
-	let backHref = '/settings/manage-connection/' + siteSlug;
+	let backHref = '/settings/manage-connection/' + site.slug;
 	if ( reason ) {
-		backHref = '/settings/disconnect-site/' + siteSlug;
+		backHref = '/settings/disconnect-site/' + site.slug;
 	}
 
-	return (
-		<div>
-			<Main className="disconnect-site__site-settings">
-				<DocumentHead title={ translate( 'Site Settings' ) } />
-				<FormattedHeader
-					headerText={ translate( 'Disable Jetpack' ) }
-					subHeaderText={ translate( "Please let us know why you're disabling Jetpack." ) }
-				/>
-				<DisconnectSurvey confirmHref={ confirmHref } />
-				<div className="disconnect-site__navigation-links">
-					<NavigationLink href={ backHref } direction="back" />
-					<NavigationLink href={ confirmHref } direction="forward" />
-				</div>
-				<Troubleshoot />
-			</Main>
-		</div>
-	);
+	if ( type === 'down' ) {
+		return (
+			<DownFlow
+				translate={ translate }
```

Comment:
Instead of passing the `translate` prop down, it's more preferrable to use the `useTranslate` hook there:
```jsx
export default function DownFlow( props ) {
  const translate = useTranslate();
  ...
}
```

This is also a great opportunity to migrate the `<DisconnectSite />` component away from the `localize` HOC, to use the `useTranslate` hook.

Reply:
done in 080554558bf4cb68c0109dcf844c8a8f555ce89f